### DISCUSSION TITLE
Enable Enhanced Domains (and Edge Network) when creating a scratch org

### DISCUSSION THREAD

**Discussion by daveespo** [forcedotcom/cli#2384] [Ideas]
Salesforce is going all-in on Enhanced Domains and they will be mandatory in Winter '23 (October 2022)

However, there is still no way to enable Enhanced Domains when creating a scratch org. Yes, you can manually go in and enable Enhanced Domains in Setup, but that's not a viable approach for CI/CD and automated testing scenarios.

For context: We have found several regressions in functionality when we enable Enhanced Domains in our orgs (mostly around Experience Cloud (f.k.a. Communities f.k.a. Customer Portal)) so it's imperative that we're able to cover our application with Enhanced Domains enabled when running a test suite.

I have tried pulling down the MyDomain Settings metadata and including it in the source push. This is what the metadata looks like when Enhanced Domains are enabled (i.e. after I enable it manually in Setup)

```
<?xml version="1.0" encoding="UTF-8"?>
<MyDomainSettings xmlns="http://soap.sforce.com/2006/04/metadata">
    <canOnlyLoginWithMyDomainUrl>false</canOnlyLoginWithMyDomainUrl>
    <doesApiLoginRequireOrgDomain>false</doesApiLoginRequireOrgDomain>
    <enableNativeBrowserForAuthOnAndroid>false</enableNativeBrowserForAuthOnAndroid>
    <enableNativeBrowserForAuthOnIos>false</enableNativeBrowserForAuthOnIos>
    <myDomainName>river-pluto-6376-dev-ed</myDomainName>
    <myDomainSuffix>MySalesforce</myDomainSuffix>
    <redirectPriorMyDomain>true</redirectPriorMyDomain>
    <use3rdPartyCookieBlockingCompatibleHostnames>true</use3rdPartyCookieBlockingCompatibleHostnames>
    <useEdge>true</useEdge>
    <useStabilizedMyDomainHostnames>true</useStabilizedMyDomainHostnames>
    <useStabilizedSandboxMyDomainHostnames>true</useStabilizedSandboxMyDomainHostnames>
</MyDomainSettings>
```

This is what it looks like when Enhanced Domains is NOT enabled

```
<?xml version="1.0" encoding="UTF-8"?>
<MyDomainSettings xmlns="http://soap.sforce.com/2006/04/metadata">
    <canOnlyLoginWithMyDomainUrl>false</canOnlyLoginWithMyDomainUrl>
    <doesApiLoginRequireOrgDomain>false</doesApiLoginRequireOrgDomain>
    <enableNativeBrowserForAuthOnAndroid>false</enableNativeBrowserForAuthOnAndroid>
    <enableNativeBrowserForAuthOnIos>false</enableNativeBrowserForAuthOnIos>
    <myDomainName>americano-latte-837-dev-ed</myDomainName>
    <myDomainSuffix>MySalesforceLimited</myDomainSuffix>
    <redirectPriorMyDomain>true</redirectPriorMyDomain>
    <use3rdPartyCookieBlockingCompatibleHostnames>true</use3rdPartyCookieBlockingCompatibleHostnames>
    <useEdge>false</useEdge>
    <useStabilizedMyDomainHostnames>true</useStabilizedMyDomainHostnames>
    <useStabilizedSandboxMyDomainHostnames>false</useStabilizedSandboxMyDomainHostnames>
</MyDomainSettings>
```

Notice the `<useEdge>`, `<useStabilizedMyDomainHostnames>`, and `<myDomainSuffix>` values are the ones that change when Enhanced Domains is enabled

In summary: Pushing the Settings metadata has no effect. It does not enable Enhanced Domains.

[a comment by an automated account is not shown]

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-11090064](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000wF9YhYAK/view)

**Comment by emilygdavis** [comment]
I agree, this would be excellent functionality to have!

[a comment by an automated account is not shown]

**Comment by mshanemc** [comment]
This'll be the default eventually, and is for some not on hyperforce https://trailhead.salesforce.com/trailblazer-community/groups/0F94S000000kF9q?tab=discussion&sort=LAST_MODIFIED_DATE_DESC and will be for others soon.

We won't do anything in the CLI to enable this--(example, if their settings/metadataAPI stuff isn't working properly).
